src/Lean/Elab/Syntax.lean

```diff
@@ -235,10 +237,25 @@ where
     | none => throwUnsupportedSyntax
 
   processNonReserved (stx : Syntax) := do
-    match stx[1].isStrLit? with
-    | some atom => return (← `(ParserDescr.nonReservedSymbol $(quote atom) false), 1)
-    | none      => throwUnsupportedSyntax
+    let some atom := stx[1].isStrLit? | throwUnsupportedSyntax
+    let id := mkCIdentFrom stx[0] ``ParserDescr.nonReservedSymbol
+    return (← `($id $(quote atom) false), 1)
 
+@[builtinTermElab parserAlias] def elabParserAlias : TermElab
+  | `(parser_alias% $id), expectedType? => do
+    let aliasName := id.getId.eraseMacroScopes
+    if (← getInfoState).enabled then
+      let decl := (← Parser.getParserAliasInfo aliasName).declName
+      if let some info := (← getEnv).find? decl then
+        pushInfoLeaf <| Info.ofTermInfo {
+          elaborator := .anonymous
+          lctx := .empty
+          expr := ← mkConst decl (info.levelParams.map mkLevelParam)
```

Comment:
That's `mkConstWithLevelParams`. Not that I'd ever expect levels to come up here.

Reply:
it's an adaptation of it to avoid the error case

Reply:
I don't see anything wrong with failing here, we'll fail below anyway in that case. But I guess it's fine this way.

Reply:
If it does `throwError`, then bad things happen. I suppose I could catch the error here, but it's not supposed to prevent the syntax from doing what it does in the case that the alias wasn't found or it had a bogus `declName` attached.

Reply:
Ah I guess that's true. Nice!